### PR TITLE
Enumerables

### PR DESCRIPTION
**Pull request by kazumaki**

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    my_each do |i|
+      return false unless yield(i)
+    end
+    true
+  end
```
Comment (wathigo) [inline enumerable.rb]: **[REQUIRED]** 
* When no block or argument is given returns true when none of the collection members are false or nil
```
Failure/Error: expect(false_array.my_all?).to be false_array.all?
     
       expected false
            got true
```
* When a class is passed as an argument returns true if all of the collection is a member of such class.
```
Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns true if all of the collection matches the Regex.
```
Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern.
```
Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+  def my_any?
+    return true unless block_given?
+
+    my_each do |i|
+      return true if yield(i)
+    end
+
+    false
+  end
```
Comment (wathigo) [inline enumerable.rb]: **[REQUIRED]** 
* When no block or argument is given returns true if at least one of the collection is not false or nil.
```
Failure/Error: expect(false_array.my_any?).to be false_array.any?
     
       expected false
            got true
```
* When a class is passed as an argument returns true if at least one of the collection is a member of such class.
```
Failure/Error: expect(array.my_any?(Integer)).to be array.any?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns false if none of the collection matches the Regex.
```
Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+  def my_none?
+    return true unless block_given?
+
+    my_each do |i|
+      return false if yield(i)
+    end
+
+    true
+  end
```
Comment (wathigo) [inline enumerable.rb]: **[REQUIRED]** 
* When no block or argument is given returns true only if none of the collection members is true.
```
Failure/Error: expect(true_array.my_none?).to be false
     
       expected false
            got true
```
* When a class is passed as an argument returns true if none of the collection is a member of such class.
```
Failure/Error: expect(array.my_none?(String)).to be true
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns true only if none of the collection matches the Regex.
```
Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
```

File: enumerable.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  def my_count(*xarg)
+    count = 0
+    if x.empty?
+      my_each { count += 1 } unless block_given?
+      my_each { |i| count += 1 if yield(i) } if block_given?
+    else
+      my_each { |i| count += 1 if i == xarg[0] }
+    end
+    count
+  end
```
Comment (wathigo) [inline enumerable.rb]: **[REQUIRED]** 
* Returns the number of items in enum through enumeration.
```
Failure/Error: expect(array.my_count).to eq array.count
      
      NameError:
        undefined local variable or method `x' for #<Array:0x000055b2df850808>
```
* Counts the number of items in enum that are equal to item if an argument is given.
```
Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
      
      NameError:
        undefined local variable or method `x' for #<Array:0x000055b2df861d88>
```
* Counts the number of elements yielding a true value if a block is given.
```
Failure/Error: expect(array.my_count(&block)).to eq array.count(&block)
      
      NameError:
        undefined local variable or method `x' for #<Array:0x000055b2df873c68>
```

File: enumerable.rb [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        ret_val = yield(ret_val, val) if i.positive?
+      end
+    else
+      ret_val = xarg[0]
+      my_each { |i| ret_val = yield(ret_val, i) }
+    end
+
+    ret_val
+  end
+end
```
Comment (wathigo) [inline enumerable.rb]: **[REQUIRED]**
* When a symbol is specified combines each element of the collection by applying the symbol as a named method.
```
Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)
      
      LocalJumpError:
        no block given (yield)
```

**Review by wathigo — CHANGES_REQUESTED** [review]
### Hi @kazumaki  :wave:  
Great work on the project.
Let's ensure that methods passes all the test cases. See the comments below :point_down:  

**After implementing the changes, submit another request for review.**

_Happy Coding :computer:  
Reviewed by: **Simon Wathigo**_

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Hi Kaz :slightly_smiling_face: 

Just one more correction to make.

- [x] In enumerables my_inject, when a symbol is specified it should combine each element of the collection by applying the symbol as a named method.
     Failure/Error: expect(range.my_inject(2, :*)).to eq range.inject(2, :*)

 - For [example](https://apidock.com/ruby/Enumerable/inject), (5..10).inject(:+)  #=> 45

Fix this and submit another form for a code review.

#### ``` Have a great day ``` :slightly_smiling_face: 

@ Uchechuwkwu Henry Inyama

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implmenting the suggested changes :clap: . However, the problem with my_inject persists, thus you need to fix it.

Try to work on the example below to fix your method and make it return true
```
range = Range.new(5,50)
puts range.my_inject(2, :*) === range.inject(2, :*) #false
```

### That's it :relaxed: . When finished, kindly resubmit another code review request :pray: 

happy coding :v: 
**Ahmed Mahfoudh**

**Review by uche-inyama — APPROVED** [review]
## Hi Kaz!

We are all good to go: :slightly_smiling_face: 

                  Finished in 0.01972 seconds (files took 0.14883 seconds to load)
                  28 examples, 0 failures.

Go ahead merge.

#### ``` Have a great day ``` :slightly_smiling_face: 

@ Uchechukwu Henry Inyama

**Pull request merged**